### PR TITLE
OP-860: Superuser type checks in users.

### PR DESCRIPTION
**Pull request by wzglinieckisoldevelo**
Ticket: https://openimis.atlassian.net/browse/OP-860
Connected to: https://github.com/openimis/openimis-be-location_py/pull/38

Changes:
- has_perms() now returns true is user is a superuser
- is_superuser checks technical user 'superuser' flag

### CODE REVIEW

File: core/models.py [hunk 1]
```diff
@@ -385,7 +385,10 @@ def is_staff(self):
 
     @property
     def is_superuser(self):
-        return False
+        if isinstance(self, TechnicalUser):
```
Comment (dborowiecki) [inline core/models.py:388]: This statement is never true, interactive user is never instance of technical user. You have to check User class (but now user doesn't have related_name, so please add it as well) And then use something like self.user.is_superuser()

File: core/models.py [hunk 2]
```diff
@@ -513,7 +516,10 @@ def is_staff(self):
 
     @property
     def is_superuser(self):
-        return self._u.is_superuser
+        if isinstance(self, TechnicalUser):
```
Comment (dborowiecki) [inline core/models.py:519]: Same here, this is never true, you have to call self.t_user.is_superuser

File: core/models.py [hunk 3]
```diff
@@ -528,10 +534,12 @@ def is_active(self):
         return True
 
     def has_perm(self, perm, obj=None):
-        i_user = self.i_user if obj is None else obj.i_user
+        self.has_perms(self, perm, obj=None)
```
Comment (dborowiecki) [inline core/models.py:537]: Don't overwrite this method in this way, has_perm is a django function and from what I see has_perms is using has perm to do the check. Therefore there will be recursion exception. You should do the check for superadmin and then return either true or super(User, self).has_perm

File: core/models.py [hunk 4]
```diff
@@ -528,10 +534,12 @@ def is_active(self):
...
+
+    def has_perms(self, perm, obj=None):
```
Comment (dborowiecki) [inline core/models.py:539]: `x if y else z` is good syntactic sugar when you have one liners in this case please change implementation to regular if else statement

**Review by dborowiecki — CHANGES_REQUESTED** [review]

File: core/models.py (hunk 4 above)
Comment (wzglinieckisoldevelo) [reply core/models.py:546]: It is artifact from reversing those changes, this code is not changed in comparison to previous version:
https://github.com/openimis/openimis-be-core_py/blob/main/core/models.py#L530

**Review by dragos-dobre — APPROVED** [review]

**Review by delcroip — APPROVED** [review]

**Pull request merged**